### PR TITLE
Fix calling llvm-nm on Windows when response files are used and input .o file is passed by relative path

### PR DESCRIPTION
**Pull request by juj**
Fix calling llvm-nm on Windows when response files are used and input .o file is specified via a relative path to a subdirectory (foo\foo.o). Add a test.

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -11409,3 +11409,31 @@ def test_gmtime_noleak(self):
     # Confirm that gmtime_r does not leak when called in isolation.
     self.emcc_args.append('-fsanitize=leak')
     self.do_other_test('test_gmtime_noleak.c')
+
+  # Test that using llvm-nm works when response files are in use, and inputs are linked using relative paths.
+  # llvm-nm has a quirk that it does not remove escape chars when printing out filenames.
```
Comment (sbc100) [inline tests/test_other.py:11414]: You can do `@with_env_modify` on the function to avoid the extra block/indentation here.

**Review by sbc100 — APPROVED** [review]
Is this bug in llvm-nm that we should file upstream perhaps?

**Comment by juj** [comment]
> Is this bug in llvm-nm that we should file upstream perhaps?

It looks like so. If you'd like, feel free to report it upstream. I have not had much success recently with upstream LLVM project reacting to my bugs..

File: tools/building.py [hunk 2]
```diff
@@ -551,6 +554,12 @@ def get_command_with_possible_response_file(cmd):
 # This function can be called either for a single file output listing ("llvm-nm a.o", or for
 # multiple files listing ("llvm-nm a.o b.o").
 def parse_llvm_nm_symbols(output):
+  # If response files are used in a call to llvm-nm, the output printed by llvm-nm has a
+  # quirk that it will contain double backslashes as directory separators on Windows. (it will
+  # not remove the escape characters when printing)
+  # Therefore canonicalize the output to single backslashes.
+  output = output.replace('\\\\', '\\')
```
Comment (sbc100) [inline tools/building.py:561]: Just to be clear, if I remove this line then `test_llvm_nm_relative_paths_works_with_response_files` will fail on windows?

**Review by sbc100 — APPROVED** [review]

File: tools/building.py (hunk 2 above)
Comment (juj) [reply tools/building.py:561]: yes

**Pull request merged**